Update pyLoad integration to async library

tr4nt0r opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Replaces the API Library, which is currently embedded into `sensor.py`, with an external async library. 

This is part of a somewhat bigger refactoring of the integration (see https://github.com/home-assistant/core/pull/115260), but I'm trying to split it up into smaller chunks. Only improved parts of the code, that i had to touch anyway, otherwise tried to keep changes as little as possible. 

External API:

https://github.com/tr4nt0r/PyLoadAPI/releases/tag/1.0.3

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Commit:
Migrate pyLoad integration to externa API library

Commit:
Add const to .coveragerc

cpfair commented on homeassistant/components/pyload/sensor.py:
```diff
@@ -109,22 +128,36 @@ def __init__(
+        except InvalidAuth:
+            _LOGGER.info("Authentication failed, trying to reauthenticate")
+            try:
+                await self.api.login()
```
It seems like this retries the login, but does not repeat the `get_status` call which originally failed. Maybe it would be cleaner if the external package performed the login internally such that clients need not worry about this retry logic at all.

tr4nt0r replied:
the next polling happens like 15 seconds later anyway, but you are right, i should either raise UpdateFailed, so that HA would show on the next polling that it recovered or call get_status again directly after the login

tr4nt0r replied:
i decided for raising UpdateFailed and letting the next polling retry to fetch the data, calling get_status again after login would also mean having to catch again all the exceptions

Commit:
raise update failed when cookie expired

bdraco committed:
Bump habluetooth to 2.5.2 (#115721)

jesserockz committed:
Bump aioesphomeapi to 24.1.0 (#115729)

emontnemery committed:
Fix stale comment in wheels.yml (#115736)

c0ffeeca7 committed:
Multiple data disks detected: tweak strings (#115713)

* Multiple data disks: tweak strings

* Fix typos

* Update homeassistant/components/hassio/strings.json

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Bump aiohttp to 3.9.5 (#115727)

changelog: https://github.com/aio-libs/aiohttp/compare/v3.9.4...v3.9.5

bdraco committed:
Bump orjson to 3.10.1 (#115728)

changelog: https://github.com/ijl/orjson/compare/3.9.15...3.10.1

emontnemery committed:
Remove stale packages from uncommenting when building wheels (#115700)

emontnemery committed:
Correct normalize_package_name (#115750)

emontnemery committed:
Remove useless any in gen_requirements_all.comment_requirement (#115751)

emontnemery committed:
Fix hassfest requirements check (#115744)

* Fix hassfest requirements check

* Add electrasmart to ignore list

bdraco committed:
Keep track of top level components (#115586)

* Keep track of top level components

Currently we have to do a set comp for icons, translations,
and integration platforms every time to split the top level
components from the platforms. Keep track of the top level
components in a seperate set so avoid having to do the setcomp
every time.

* remove impossible paths

* remove unused code

* preen

* preen

* fix

* coverage and fixes

* Update homeassistant/core.py

* Update homeassistant/core.py

* Update tests/test_core.py

emontnemery committed:
Improve standard library violation check in hassfest (#115752)

* Improve standard library violation check in hassfest

* Improve prints

* Improve error message

epenet committed:
Bump renault-api to 0.2.2 (#115738)

tomaszsluszniak committed:
Add Sanix integration (#106785)

* Add Sanix integration

* Add Sanix integration

* Add sanix pypi package

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Add Sanix integration

* Fix ruff

* Fix

* Fix

---------

Co-authored-by: Joostlek <joostlek@outlook.com>

emontnemery committed:
Fix flaky qld_bushfire test (#115757)

emontnemery committed:
Fix homeworks import flow (#115761)

emontnemery committed:
Allow [##:##:##] type keypad address in homeworks (#115762)

Allow [##:##:##] type keypad address

autinerd committed:
Add config flow to enigma2 (#106348)

* add config flow to enigma2

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* fix suggested change

* use parametrize for config flow tests

* Restore PLATFORM_SCHEMA and add create_issue to async_setup_platform

* fix docstring

* remove name, refactor config flow

* bump dependency

* remove name, add verify_ssl, use async_create_clientsession

* use translation key, change integration type to device

* Bump openwebifpy to 4.2.1

* cleanup, remove CONF_NAME from entity, add async_set_unique_id

* clear unneeded constants, fix tests

* fix tests

* move _attr_translation_key out of init

* update test requirement

* Address review comments

* address review comments

* clear strings.json

* Review coments

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bdraco committed:
Avoid linear search to remove labels and floors from area registry (#115675)

* Avoid linear search to remove labels and floors from area registry

* simplify

bdraco committed:
Avoid linear search to clear labels and areas in the device registry (#115676)

bdraco committed:
Avoid linear search to remove a label from the entity registry (#115674)

* Avoid linear search to remove a label from the entity registry

* simplify

bdraco committed:
Reduce duplicate code in the device registry (#115677)

mdegat01 committed:
Add repairs to hassio manifest (#115486)

* Add repairs to hassio manifest

* Remove unnecessary fixture

joostlek committed:
Remove obsolete translation keys in Sanix (#115764)

emontnemery committed:
Include hash of requirements.txt in venv cache key (#115759)

bdraco committed:
Bump habluetooth to 2.6.0 (#115724)

bdraco committed:
Simplify config_entries entity registry filter (#115678)

CFenner committed:
Add secondary temperature sensor for DHW in ViCare (#106612)

* add temp2 sensor

* Update strings.json

Noltari committed:
Allow selecting Air Quality mode for Airzone Cloud (#106769)

* airzone_cloud: add Air Quality mode select

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* trigger CI

* trigger CI

* Update select.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* airzone_cloud: select: remove AirzoneSelectDescriptionMixin usage

* airzone_cloud: select: rename AIR_QUALITY_DICT

---------

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

emontnemery committed:
Use aiohttp-zlib-ng[isal] (#115767)

LaStrada committed:
Fix unrecoverable error when fetching airthings_ble data (#115699)

janiversen committed:
Cleanup modbus test mocks (#115412)

isXander committed:
Guard negative values for IPP states (#107446)

* Guard negative values for IPP states

* ruff format

* Update sensor.py

---------

Co-authored-by: Chris Talkington <chris@talkingtontech.com>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

bdraco committed:
Replace aiohttp-zlib-ng[isal] with aiohttp-isal (#115777)

* Replace aiohttp-zlib-ng[isal] with aiohttp-isal

The extra was causing wheel builds to fail

Since isal works on all of our supported platforms we can
always use it and drop the need for zlib-ng

https://github.com/home-assistant/core/actions/runs/8725019072

https://github.com/bdraco/aiohttp-isal

* typo

jbouwh committed:
Add mqtt notify platform (#115653)

* Add mqtt notify platform

* Stale docstring

autinerd committed:
Address late reviews for the enigma2 config flow (#115768)

* Address late reviews for the enigma2 config flow

* fix tests

* review comments

* test for issues

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Add sanix sensor tests (#115763)

Add sanix tests

allenporter committed:
Bump google-nest-sdm to 3.0.4 (#115731)

bdraco committed:
Simplify labels and areas template calls (#115673)

The labels and areas are already exposed on the object

felipediel committed:
Bump Broadlink to 0.19.0 (#115742)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Migrate snooze config flow to use eager_start (#115658)

bdraco committed:
Bump habluetooth to 2.7.0 (#115783)

bdraco committed:
Bump bluetooth-auto-recovery to 1.4.1 (#115792)

kbx81 committed:
Add valve entity support for ESPHome (#115341)

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

bdraco committed:
Fix failing sanix tests (#115793)

Fixing failing sanix tests

Regenerate snapshot to match what actually happens. There is no
translation keys for these two

dependabot[bot] committed:
Bump github/codeql-action from 3.25.0 to 3.25.1 (#115796)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

krzysztof-kwitt committed:
Bump connect-box to 0.3.1 (#107852)

strawgate committed:
Ambient Weather: Check for key existence before checking value (#115776)

MartinHjelmare committed:
Add script to compare alexa locales with upstream (#114247)

* Add script to compare alexa locales with upstream

* Use a function in script

* Add test base

* Complete output assertion

* Add type annotation

* Add note to docstring

* Update script call example

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

---------

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

mkmer committed:
Handle connection error in honeywell (#108168)

* Handle connection error

* Catch connection error

* Add tests

* Add translation strings

* Clean up overlapping exceptions

* ServiceValidationError

* HomeAssistant Error translations

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

luca-angemi committed:
Add state class to mobile_app restore entity (#115798)

add state class

erwindouna committed:
Add diagnostics platform to DSMR Reader (#115805)

* Add diagnostics platform

* Feedback

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jbouwh committed:
Translate service validation errors (#115024)

* Move service validation error message to translation cache

* Fix test

* Revert unrelated change

* Address review comments

* Improve error message

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

vexofp committed:
Clarify cover toggle logic; prevent opening when already open (#107920)

Co-authored-by: Erik Montnemery <erik@montnemery.com>

ptashek committed:
Add support for payload_template in rest component (#107464)

* Add support for payload_template in rest component

* Update homeassistant/components/rest/schema.py

* Update homeassistant/components/rest/data.py

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

mwielgoszewski committed:
Add hvac_action attribute to iAqualink Thermostat climate entities (#107803)

* Update climate.py

* Reorder if/else statements per @dcmeglio's suggestion

* Don't infer state, actually read it from the underlying device

* HVACAction has a HEATING state, not ON

* Update homeassistant/components/iaqualink/climate.py

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

vanbalken committed:
Bump arris-tg2492lg to 2.2.0 (#107905)

Bumps arris-tg2492lg from 1.2.1 to 2.2.0

bdraco committed:
Bump habluetooth to 2.8.0 (#115789)

* Bump habluetooth to 2.8.0

Adds support for recovering some adapters that fail to initialize
due to kernel races

* bump lib

* tweak

bdraco committed:
Add bluetooth adapter model and manufacturer to config flow (#115780)

* Show bluetooth adapter model and manufacturer in config flow

If there are multiple adapters, it could be a bit difficult
to figure out which one is which

* Show bluetooth adapter model and manufacturer in config flow

If there are multiple adapters, it could be a bit difficult
to figure out which one is which

* reorder

* reorder

* names

* remove

* fix incomplete mocking

* more missing mocks

bdraco committed:
Add reauth support to oncue (#115667)

* Add reauth support to oncue

* review comments

* reauth on update failure

* coverage

bdraco committed:
Skip processing websocket_api schema if it has no arguments (#115618)

* Skip processing websocket_api schema if has no arguments

About 40% of the websocket commands on first connection have
no arguments. We can skip processing the schema for these cases

* cover

* fixes

* allow extra

* Revert "allow extra"

This reverts commit 85d9ec36b30aa2aedecd8571c7ed734d0b0a9b05.

* match behavior

bdraco committed:
Small cleanups to translation loading (#115583)

- Add missing typing
- Convert a update loop to a set comp
- Save some indent

bdraco committed:
Migrate device_sun_light_trigger to use async_track_state_change_event (#115555)

* Migrate device_sun_light_trigger to use async_track_state_change_event

async_track_state_change is legacy and will eventually be deprecated after
all core usage is removed. There are only two places left

* coverage

bdraco committed:
Deprecate async_track_state_change in favor of async_track_state_change_event (#115558)

albertomontesg reviewed (commented):
Small nit, but for the rest looks good

tr4nt0r commented on homeassistant/components/pyload/sensor.py:
```diff
@@ -22,22 +25,22 @@
-DEFAULT_PORT = 8000
 
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+SCAN_INTERVAL = timedelta(seconds=15)
```
But that's how usually the polling interval is defined

https://developers.home-assistant.io/docs/integration_fetching_data#separate-polling-for-each-individual-entity

brg468 committed:
Add battery binary sensor to Rachio hose timer (#115810)

Co-authored-by: J. Nick Koston <nick@koston.org>

orevron committed:
Add zhimi.fan.za3 to xiaomi_miio workaround unable to discover device (#108310)

* add  zhimi.fan.za3 to workaround fix unable to discover issue

* Update __init__.py

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Commit:
Add support for room sensor accessories assigned to a Honeywell (Lyric) Thermostat (#104343)

* Add support for room sensor accessories.

- Update coordinator to refresh and grab information about room sensor accessories assigned to a thermostat
- Add sensor entities for room humidity and room temperature
- Add devices to the registry for each room accessory
- "via_device" these entities through the assigned thermostat.

* fixed pre-commit issues.

* PR suggestions

- update docstring to reflect ownership by thermostat
- fixed potential issue where a sensor would not be added if its temperature value was 0

* fix bad github merge

* asyicio.gather futures for updating theromstat room stats

bdraco committed:
Handle Bluetooth adapters in a crashed state (#115790)

* Skip bluetooth discovery for Bluetooth adapters in a crashed state

* fixes

* fixes

* adjust

* coverage

* coverage

* fix race

bdraco commented on homeassistant/components/pyload/sensor.py:
```diff
@@ -77,16 +80,32 @@ def setup_platform(
-        return
+        await pyloadapi.login()
+    except CannotConnect as conn_err:
+        raise ConfigEntryNotReady(
```
This isn't setup via a config entry so you need to use `PlatformNotReady` instead

tr4nt0r replied:
change it

bdraco commented on homeassistant/components/pyload/sensor.py:
```diff
@@ -77,16 +80,32 @@ def setup_platform(
+            translation_key="parse_exception",
+        ) from e
+    except InvalidAuth as e:
+        raise ConfigEntryAuthFailed(
```
Since this isn't a config entry this won't work

`ConfigEntryAuthFailed` starts `async_step_reauth` in a config flow but there is no config flow here.

tr4nt0r replied:
yeah, sorry, config flow comes with another PR ^^ 
So the right exception here would also be `PlatformNotReady` ?

Commit:
fix exceptions

matrixd2 committed:
Bump yolink-api to 0.4.3 (#115794)

bieniu committed:
Bump `accuweather` to version 3.0.0 (#115820)

Bump accuweather to version 3.0.0

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

Kane610 committed:
Bump aiounifi to v75 (#115819)

finder39 committed:
Add get_torrents service to qBittorrent integration (#106501)

* Upgrade QBittorrent integration to show torrents

This brings the QBittorrent integration to be more in line with the Transmission integration. It updates how the integration is written, along with adding sensors for Active Torrents, Inactive Torrents, Paused Torrents, Total Torrents, Seeding Torrents, Started Torrents.

* Remove unused stuff

* Correct name in comments

* Make get torrents a service with a response

* Add new sensors

* remove service

* Add service with response to get torrents list

This adds a service with a response to be able to get the list of torrents within qBittorrent

* update

* update from rebase

* Update strings.json

* Update helpers.py

* Update to satisfy lint

* add func comment

* fix lint issues

* another update attempt

* Fix helpers

* Remove unneccesary part in services.yaml and add translations

* Fix return

* Add tests

* Fix test

* Improve tests

* Fix issue from rebase

* Add icon for get_torrents service

* Make get torrents a service with a response

* remove service

* Add service with response to get torrents list

This adds a service with a response to be able to get the list of torrents within qBittorrent

* Update to satisfy lint

* Handle multiple installed integrations

* fix lint issue

* Set return types for helper methods

* Create the service method in async_setup

* Add CONFIG_SCHEMA

* Add get_all_torrents service

* fix lint issues

* Add return types and ServiceValidationError(s)

* Fix naming

* Update translations

* Fix tests

hahn-th committed:
Show default profiles in homematic cloud climate entity (#107348)

* Default names for visible profiles

* Increase number of devices in test

* remove unnecessary check

* Add testcase and split another into two

* Add type annotations and docstring

* Remove code which not belongs to the PR

* Add myself to codeowners

bdraco committed:
Bump aiodiscover to 2.1.0 (#115823)

dependabot[bot] committed:
Bump actions/upload-artifact from 4.3.1 to 4.3.2 (#115842)

dependabot[bot] committed:
Bump actions/download-artifact from 4.1.4 to 4.1.5 (#115841)

jbouwh committed:
Add group tests with mixed domain entities (#115849)

bieniu committed:
Bump `gios` to version 4.0.0 (#115822)

Bump gios to version 4.0.0

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

chemelli74 committed:
Add missing media_player features to Samsung TV (#115788)

* add missing features

* fix snapshot

joostlek committed:
Add reauth flow to Google Tasks (#109517)

* Add reauth flow to Google Tasks

* Update homeassistant/components/google_tasks/config_flow.py

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

* Add tests

* Reauth

* Remove insta reauth

* Fix

---------

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

joostlek committed:
Make Withings recoverable after internet outage (#115124)

avee87 committed:
Fix Hyperion light not updating state (#115389)

slyoldfox committed:
Add scheduled mode to renault charge mode (#115427)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

bdraco committed:
Ensure scripts with timeouts of zero timeout immediately (#115830)

jbouwh committed:
Fix mutable objects in group registry class (#115797)

bieniu committed:
Bump `nextdns` to version 3.0.0 (#115854)

Bump nextdns to version 3.0.0

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bieniu committed:
Bump `nettigo_air_monitor` to version 3.0.0 (#115853)

Bump nettigo_air_monitor to version 3.0.0

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bdraco committed:
Bump bluetooth-adapters to 0.19.0 (#115864)

mib1185 committed:
Automatic cleanup of entity and device registry in AVM FRITZ!SmartHome (#114601)

bieniu committed:
Use snapshot testing in NextDNS (#115879)

* Use snapshot testing in NextDNS sensor

* Use snapshot testing in NextDNS switch

* Use snapshot testing in NextDNS binary sensor

* Use snapshot testing in NextDNS button

---------

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bieniu committed:
Use snapshot testing in GIOS sensor (#115876)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bieniu committed:
Use snapshot testing in Brother sensor (#115875)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bieniu committed:
Use snapshot testing in NAM sensor and diagnostics (#115877)

* Use snapshot testing in NAM diagnostics

* Use snapshot testing in NAM sensor

* Add NAM data fixture

---------

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

jjlawren committed:
Bump plexapi to 4.15.12 (#115872)

natekspencer committed:
Bump pylitterbot to 2023.5.0 (#115856)

joostlek committed:
Add test helper to snapshot a platform (#115880)

* Add test helper to snapshot a platform

* Add test helper to snapshot a platform

joostlek committed:
Use snapshot test helper in Brother (#115885)

joostlek committed:
Use snapshot test helper in Accuweather (#115884)

bieniu committed:
Use snapshot test helper in GIOS (#115893)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bieniu committed:
Use snapshot test helper in NAM (#115894)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bieniu committed:
Use snapshot test helper in NextDNS (#115895)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

EnjoyingM committed:
Fix Wolf Smart Set Authentication and Session Management (#115815)

* Fix Wolf Smart Set Authentication and Session Management

Fix in the library to respect Wolf API token lifetime and implement Session Management

* Updatie requirments

* Update Code Owner

joostlek committed:
Use snapshot test helper in AO Smith (#115890)

albertomontesg committed:
Make release channel a hardcoded enum rather than a free form string (#115595)

* Make release channel a hardcoded enum rather than a free form string

* Update enum comparison to remove equality and us identity comparison

* Fix comparison condition to match the previous implementation

* Update tests to use Enum instead of string

joostlek committed:
Use snapshot test helper in Analytics insights (#115889)

joostlek committed:
Use snapshot test helper in Ambient Network (#115887)

autinerd committed:
Bump ruff to 0.4.1 (#115873)

r-binder committed:
Add tls support for AVM Fritz!Tools (#112714)

Bre77 committed:
Fix sensor entity description in Teslemetry (#115614)

Add description back to sensor entity

bdraco committed:
Fix flaky history stats test (#115824)

Lash-L committed:
Fix Roborock status not correctly mapping for some devices (#115646)

Use device_info.model instead of name

zxdavb committed:
Improve debug logging for evohome (#110256)

better logging

agoode committed:
Add test for prometheus export of entities becoming unavailable and available again (#112157)

Add test for state change to unavailable and back

jbouwh committed:
Improve service validation exception test and translation key (#115843)

* Small improvement to service validation exception test and translation key

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Refactor string assertion

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

wittypluck committed:
Allow manual delete of stale Unifi device from UI (#115267)

* Allow manual delete of stale device from UI

* Add unit tests for remove_config_entry_device

mib1185 committed:
Fix geo location attributes of Tankerkoenig sensors (#115914)

* geo location attributes needs to be float

* make mypy happy

joostlek committed:
Refactor Totalconnect binary sensor (#115629)

Kane610 committed:
Remove sensor exposing UniFi WLAN password (#115929)

austinmroczek committed:
Bump total_connect_client to 2023.12.1 (#115928)

bump total_connect_client to 2023.12.1

mkmer committed:
Add error translations to Blink (#115924)

* Add translations
Catch timeout in snap

* Grammer cleanup

mib1185 committed:
Make use of snapshot testing in Synology DSM (#115931)

bdraco committed:
Convert MQTT to use asyncio (#115910)

bdraco committed:
Temporarily pickup mqtt codeowner (#115934)

autinerd committed:
Enable Ruff A001 (#115654)

joostlek committed:
Introduce base entity for totalconnect (#115936)

allenporter committed:
Bump ical to 8.0.0 (#115907)

Co-authored-by: J. Nick Koston <nick@koston.org>

andarotajo committed:
Add optional location based region to dwd_weather_warnings (#96027)

* Add device tracker option

* Update const name to be more understandable

* Clean up sensor code

* Clean up init and coordinator

* Add tests and update util function and it's usage

* Switch to using the registry entry and add tests

* Clean up code

* Consolidate duplicate code and adjust tests

* Fix runtime error

* Fix blocking of the event loop

* Adjust API object handling

* Update homeassistant/components/dwd_weather_warnings/exceptions.py

* Optimize coordinator data update

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

rappenze committed:
Add type hints in fibaro device (#106874)

* Add typings in fibaro device

* Fix type hints

* Fix type hints

* Remove unused method parameter

* Improve log message

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

rappenze committed:
Refactor fibaro connect (#106875)

* Refactor fibaro connect

* Remove obsolete test

* Add comment about ignored return value

dependabot[bot] committed:
Bump actions/checkout from 4.1.2 to 4.1.3 (#115945)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Quentame committed:
Add Epic Games Store integration (#104725)

* Add Epic Games Store integration

Squashed commit of the following PR: #81167

* Bump epicstore-api to 0.1.7 as it handle better error 1004

Thanks to https://github.com/SD4RK/epicstore_api/commit/d7469f7c99508c06b3867fecbcf291ebf86c4c72

* Use extra_state_attributes instead of overriding state_attributes

* Review: change how config_flow.validate_input is handled

* Use LanguageSelector and rename locale to language

* Review: init-better use of hass.data.setdefault

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Review: don't need to update at init

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Revert "Review: don't need to update at init" --> not working otherwise

This reverts commit 1445a87c8e9b7247f1c9835bf2e2d7297dd02586.

* Review: fix config_flow.validate_input/retactor following lib bump

* review: merge async_update function with event property

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* hassfest

* Fix duplicates data from applied comment review 5035055

* review: thanks to 5035055 async_add_entities update_before_add param is not required anymore

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Fix Christmas special "Holiday sale" case

* gen_requirements_all

* Use CONF_LANGUAGE from HA const

* Move CalendarType to const

* manifest: integration_type -> service

Co-authored-by: Sid <27780930+autinerd@users.noreply.github.com>

* calendar: remove date start/end assert

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* const: rename SUPPORTED_LANGUAGES

* hassfest

* config: Move to ConfigFlowResult

* coordinator: main file comment

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* ruff & hassfest

* review: do not guess country

* Add @hacf-fr as codeowner

* review: remove games extra_attrs
Was dropped somehow:
- 73c20f34803b0a0ec242bf0740494f17a68f6f59 review: move games extra_attrs to data service
- other commit that removed the service part

* review: remove unused error class
was removed:
- 040cf945bb5346b6d42b3782b5061a13fb7b1f6b

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Sid <27780930+autinerd@users.noreply.github.com>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

jesserockz committed:
Bump aioesphomeapi to 24.2.0 (#115943)

joostlek committed:
Introduce base location entity for totalconnect (#115938)

* Introduce base location entity for totalconnect

* Update homeassistant/components/totalconnect/entity.py

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

---------

Co-authored-by: TheJulianJES <TheJulianJES@users.noreply.github.com>

bieniu committed:
Move NextDNS data update coordinators to the coordinator module (#115919)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

timmo001 committed:
Update ovoenergy to 2.0.0 (#115921)

Co-authored-by: J. Nick Koston <nick@koston.org>

joostlek committed:
Add snapshot tests to Totalconnect (#115952)

* Add snapshot tests to Totalconnect

* Add snapshot tests to Totalconnect

jesserockz committed:
ESPHome: Add datetime entities (#115942)

mib1185 committed:
Address late review for AVM Fritz!Smarthome (#115960)

fix typo

marcolivierarsenault committed:
Reduce ecobee throttle (#115968)

reduce ecobee throttle

bdraco committed:
Improve reliability of homeassistant_alerts updates (#115974)

bdraco committed:
Small fixes for processing integration requirements (#115973)

dknowles2 committed:
Remove YAML support from Hydrawise (#115966)

rajlaud committed:
Use start helper in squeezebox for server discovery (#115978)

emontnemery committed:
Generate requirements per supported architecture (#115708)

* Generate requirements per supported architecture

* Don't store wheels requirements in the repo

* Dry run

* Set Python version

* Install base packages

* Fix

* Fix

* Fix

* Fix typo

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Genarate requirements_all_pytest.txt

* Fix hassfest

* Reenable building wheels

* Remove unneeded code

* Address review comment

* Fix lying comment

* Add tests, address review comments

* Deduplicate

* Fix file name

* Add comment

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

marcolivierarsenault committed:
Revert "Reduce ecobee throttle (#115968)" (#115981)

bdraco committed:
Migrate MQTT EnsureJobAfterCooldown to use eager start (#115977)

karwosts committed:
Make recorder.purge_entities require at least one entity filter value (#110066)

Co-authored-by: J. Nick Koston <nick@koston.org>

gjohansson-ST committed:
Bump holidays to 0.47 (#115992)

kimdv committed:
Add bandwidth sensor for unifi device ports (#115362)

Kane610 committed:
Bump aiounifi to v76 (#116005)

* Bump aiounifi to v76

edenhaus committed:
Bump uv to 0.1.35 (#115985)

gjohansson-ST committed:
Add config flow to Folder Watcher (#105605)

* Add config flow to Folder Watcher

* Add tests config flow

* docstrings

* watcher is sync

* Fix strings

* Fix

* setup_entry issue

* ConfigFlowResult

* Review comments

* Review comment

* ruff

* new date

thomaskistler committed:
Update Hydrawise from the legacy API to the new GraphQL API (#106904)

* Update Hydrawise from the legacy API to the new GraphQL API.

* Cleanup

shaiu committed:
Add coordinator to 17Track (#115057)

* Add coordinator to 17Track

* Add coordinator to 17Track

remove SensorEntityDescription (different PR)

* Update homeassistant/components/seventeentrack/coordinator.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/seventeentrack/sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Add coordinator to 17Track

fix CR

* Add coordinator to 17Track

fix second CR

* Add coordinator to 17Track

remove commented out code + fix display name

* Add coordinator to 17Track

created a set outside _async_create_remove_entities function

* Add coordinator to 17Track

fix CR

* Add coordinator to 17Track

fix CR 2

* Update homeassistant/components/seventeentrack/coordinator.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Add coordinator to 17Track

raise UpdateFailed if API throws an exception

* Add coordinator to 17Track

merge calls

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

dependabot[bot] committed:
Bump actions/download-artifact from 4.1.4 to 4.1.6 (#116017)

dependabot[bot] committed:
Bump actions/upload-artifact from 4.3.1 to 4.3.3 (#116015)

joostlek committed:
Use generator expression in totalconnect (#116020)

myhomeiot committed:
Improve Vodafone Station empty/unavailable phone number detection (#115696)

Vodafone Sercomm H300S model incorrectly reports phone_unavailable1/phone_unavailable2 flags.

Commit:
Don't raise errors when using datetime objects in `as_datetime` Jinja function/filter (#109062)

* add support for datetime objects to as_datetime

* change import of datetime.date

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

janiversen committed:
Deprecate modbus:restart service (#115754)

cdce8p committed:
Add uv version to wheels cache key [ci] (#116021)

jesserockz committed:
Bump aioesphomeapi to 24.3.0 (#116004)

joostlek committed:
Add event platform to Lutron (#109121)

* Add event platform to Lutron

* Add event platform to Lutron

* Fix

* Fix

* Fix

* Add deprecation note

* Fix

* Fix

* Update homeassistant/components/lutron/event.py

* Update homeassistant/components/lutron/event.py

* Fix

edenhaus committed:
Bump deebot-client to 7.0.0 (#116025)

Thomas55555 committed:
Add number platform to Husqvarna Automower (#115125)

* Add number platform to Husqvarna Automower

* use fixture to enable by default

* replace state test with snapshot test

* make property in entity description

* send value as integer

* give the exists functions something to do

catsmanac committed:
Add additional zeroconf discovery coverage and logging to enphase_envoy (#114405)

* add debug info to zeroconf for enphase_envoy

* Implement review feedback, lost space

Co-authored-by: Charles Garwood <cgarwood@newdealmultimedia.com>

* review feedback textual changes.

* implement review feedbackw.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Add some more zeroconf tests and valid jwt

* review feedback assert abort reason and keyerror for serialnumber

* Review feedback config flow test ends with abort or create_entry

* Review feedback optimize resource usage

* Cover new code in test.

* Use caplog for debug COV

---------

Co-authored-by: Charles Garwood <cgarwood@newdealmultimedia.com>
Co-authored-by: J. Nick Koston <nick@koston.org>

edenhaus committed:
Use location name on self hosted Ecovacs config entries (#115294)

LaStrada committed:
Fix KeyError error when fetching sensors (Airthings) (#115844)

cdce8p committed:
Fix invalid tuple annotations (#116035)

Spacetech committed:
Expose dynamic range status in Onkyo media player (#109099)

Expose HDR status in Onkyo media player

Co-authored-by: Erik Montnemery <erik@montnemery.com>

cereal2nd committed:
Velbus Cover: Assume state for VMBxBL modules (#109213)

VolkerStolz committed:
Augment SyntaxError raised during dependency collection with offending filename (#109204)

* Capture parsing exception when collecting dependencies and augment with offending filename.

Whereas before any syntax error in some component-file would result in an opaque SyntaxError without pointing out the file, now the result will show as:

```
  File "/usr/local/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/pool.py", line 873, in next
    raise value
SyntaxError: Can't parse file homeassistant/components/your/file.py
```

* tweak

* D'oh, had pre-commit hook still off.

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

cdce8p committed:
Remove unnecessary type ignores (#116036)

cdce8p committed:
Improve energy typing (#116034)

Thomas55555 committed:
Use snapshot test helper in Husqvarna Automower (#116039)

mib1185 committed:
Bump python-fritzhome to 0.6.11 (#115904)

dknowles2 committed:
Remove platform schema from Hydrawise (#116032)

emontnemery committed:
Include libgammu-dev in devcontainer (#115983)

mill1000 committed:
Add fan preset translations and icons to BAF (#109944)

malkstar committed:
Allow plain text messages in telegram_bot (#110051)

* Add new plain_text parser

Passing None in the parse_mode kwargs on the various bot methods actually means that no parser is used.

* Add new plain text parser option to services.yaml

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Commit:
Add tests

tr4nt0r commented:
oh shit, something went wrong with the rebase, i'm sorry /o\

bdraco commented:
No worries it happens. Please start a fresh PR as there are now 76 people subscribed to this one.
